src/firefly/js/templates/lightcurve/LcManager.js

```diff
@@ -100,19 +144,44 @@ export function* lcManager(params={}) {
             case TBL_RESULTS_ACTIVE :
                 newLayoutInfo = handleTableActive(newLayoutInfo, action);
                 break;
+            case VALUE_CHANGE:
+                if (['time', 'flux', 'periodMin', 'periodMax'].includes(action.payload.fieldKey)) {
+                    // re-validate period value
+                    if (action.payload.fieldKey.startsWith('period')) { // periodMin or periodMax is changed
+                        const fields = FieldGroupUtils.getGroupFields(LC.PERIOD_FINDER);
+                        const per = fields && get(fields, ['period', 'value']);
+
+                        dispatchValueChange({
+                            fieldKey: lowerCase(LC.PERIOD_CNAME),
+                            groupKey: LC.PERIOD_FINDER,
+                            value: per
+                        });
+                    }
+                }
+                break;
         }
 
         if (newLayoutInfo !== layoutInfo) {
             dispatchUpdateLayoutInfo(newLayoutInfo);
-
+        }
+        if (bInitPeriod) {
+            periodPageMode(LC.PERIOD_PAGE);
```

Comment:
dispatchUpdateLayoutInfo is preferred to be called first in order to get better visual effect for  the page transition.